src/zenml/integrations/seldon/seldon_client.py

```diff
@@ -0,0 +1,735 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+
+import json
+import time
+from typing import Dict, List, Optional
+
+from kubernetes import client as k8s_client
+from kubernetes import config as k8s_config
+from pydantic import BaseModel, Field, ValidationError
+
+from zenml.logger import get_logger
+from zenml.utils.enum_utils import StrEnum
+
+logger = get_logger(__name__)
+
+
+SELDON_DEPLOYMENT_KIND = "SeldonDeployment"
+SELDON_DEPLOYMENT_API_VERSION = "machinelearning.seldon.io/v1"
+
+
+class SeldonDeploymentMetadata(BaseModel):
+
+    name: str
+    labels: Dict[str, str] = Field(default_factory=dict)
+
+    class Config:
+        """Pydantic configuration class."""
+
+        # validate attribute assignments
+        validate_assignment = True
+        # Ignore extra attributes from the CRD that are not reflected here
+        extra = "ignore"
+
+
+class SeldonDeploymentPredictiveUnitType(StrEnum):
+
+    UNKNOWN_TYPE = "UNKNOWN_TYPE"
+    ROUTER = "ROUTER"
+    COMBINER = "COMBINER"
+    MODEL = "MODEL"
+    TRANSFORMER = "TRANSFORMER"
+    OUTPUT_TRANSFORMER = "OUTPUT_TRANSFORMER"
+
+
+class SedonDeploymentPredictiveUnit(BaseModel):
```

Comment:
Typo in classname: Se**l**donDeploymentPredictiveUnit

Reply:
Good catch ! Thanks !